openqa-trigger-bisect-jobs

```diff
@@ -219,35 +257,50 @@ def main(args):
     log.debug("Received job data: %s" % test_data)
     test = job["settings"]["TEST"]
     log.debug("Found test name '%s'" % test)
+
     created = ""
     pattern = re.compile(r"Created\sjob\s.* (http.+\/t\d+)$")
+    added = []
     for key in all_changes:
         changes = all_changes[key]
-        removed, added = list(changes[GOOD] - changes[BAD]), list(
+        removed_key, added_key = list(changes[GOOD] - changes[BAD]), list(
             changes[BAD] - changes[GOOD]
         )
-        log.debug("[%s] removed: %s, added: %s" % (key, removed, added))
-        for issue in sorted(added):
-            log.info(
-                "[%s] Triggering one bisection job without issue '%s'" % (key, issue)
-            )
-            new = ",".join(sorted(list(changes[BAD] - set([issue]))))
-            log.debug("New set of %s='%s'" % (key, new))
-            test_name = test + ":investigate:bisect_without_%s" % issue
-            out = openqa_clone(
-                [
-                    args.url,
-                    key + "=" + new,
-                    "TEST=" + test_name,
-                    "OPENQA_INVESTIGATE_ORIGIN=" + args.url,
-                    "MAINT_TEST_REPO="
-                ],
-                args.dry_run,
+        log.debug("[%s] removed: %s, added: %s" % (key, removed_key, added_key))
+        added += added_key
+
+    # whole sort is to simplify testability of code
+    for issue in sorted(list({i.incident_id for i in added}), key=int):
```

Comment:
The comment already says this is for the benefit of unit tests. Which might be a bit wasteful if it's otherwise not needed. Did you look into alternatives? It kind of looks like all this does is order the log messages. Maybe they can be sorted afterwards?

Reply:
I looked , unfortuanetly `MagicMock` from `unittest` excepts exact values in assert so we are here

btw. If you look to original code --> it has more sorts only for sake of unit tests

Reply:
The sorts are for unit tests, but also for humans. How many of those issues do we have at a maximum? 30? Then the sort will not make any visible difference to the total runtime of the bisect script.